### ISSUE TITLE
Shader (not visual) workflow awkward with external editing

### ISSUE THREAD

**Issue by Vultraz** [godotengine/godot#27892]
When working with plain `.shader` files, you have to close and reopen the whole project in order for any externally made changes to take effect. For example, say you make a change to your shader you then want to revert. Even if you go into git and revert your local changes, they remain in Godot until you close the whole project. Since the shaders open in the bottom dock, there's no way to "close" that tab, refresh it, or anything.

Basically, it makes the shader workflow very annoying if you're using a VCS or another external editor like VS Code and just happen to have opened the shader file in Godot.

**Comment by akien-mga** [comment]
Duplicate of #20250.
